### DISCUSSION TITLE
How can I use mongodb $exists filter on strapi query?

### DISCUSSION THREAD

**Discussion by darron1217** [strapi/strapi#8068] [Legacy Topics]
How can I use mongodb $exists filter on strapi query?
(On date type field)

I tried `_null` filter, but mongoose emits `Invalid format` error.

Here's my code
```js
    const classCount = await strapi.query('class').count({
      ...defaultFilter,
      startDate_lte: today.format('YYYY-MM-DD'),
      _or: [
        {endDate_gte: today.format('YYYY-MM-DD')},
        {endDate_null: true}
      ],
      isActive: true
    })
```

**Comment by derrickmehaffy** [comment]
For mongoose you will need to use the custom query: https://strapi.io/documentation/v3.x/concepts/queries.html#mongoose

**Reply by darron1217** [reply]
Okay then, how about fixing `_null` filter on date?
Mongodb document says that `null` filter also checks existance of field.
https://docs.mongodb.com/manual/tutorial/query-for-null-fields/

**Reply by darron1217** [reply]
Oh, I found PR of this issue.
https://github.com/strapi/strapi/pull/7801

I hope this to be merged soon 👍

**Reply by derrickmehaffy** [reply]
Merged but won't be released until v3.1.7 is out and we don't have an ETA at the moment
